pkgs/servers/http/nginx/modules.nix

```diff
@@ -67,15 +67,14 @@ in
   };
 
   brotli = {
-    src = let gitsrc = pkgs.fetchFromGitHub {
-      name = "brotli";
+    src = let src' = fetchFromGitHub {
       owner = "google";
       repo = "ngx_brotli";
-      rev = "25f86f0bac1101b6512135eac5f93c49c63609e3";
-      sha256 = "02hfvfa6milj40qc2ikpb9f95sxqvxk4hly3x74kqhysbdi06hhv";
+      rev = "6e975bcb015f62e1f303054897783355e2a877dc";
+      sha256 = "sha256-G0IDYlvaQzzJ6cNTSGbfuOuSXFp3RsEwIJLGapTbDgo=";
     }; in
-      pkgs.runCommand "ngx_brotli-src" { } ''
-        cp -a ${gitsrc} $out
+      pkgs.runCommand "brotli" { } ''
+        cp -a ${src'} $out
         substituteInPlace $out/filter/config \
           --replace '$ngx_addon_dir/deps/brotli/c' ${lib.getDev pkgs.brotli}
       '';
```

Comment:
Let's use `postFetch` for this.

Reply:
this variant not workin:
```
...
      rev = "6e975bcb015f62e1f303054897783355e2a877dc";
      sha256 = "sha256-nBstePpxUp58ebDpSlCWmhkCtiRlJsSDnagLkzxMZhE=";
      postFetch = ''
        substituteInPlace $out/filter/config \
          --replace '$ngx_addon_dir/deps/brotli/c' ${lib.getDev pkgs.brotli}
      '';
    };
    inputs = [ pkgs.brotli ];
  };
```
Build log:
```
nginx> checking for getaddrinfo() ... found
nginx> configuring additional modules
nginx> adding module in /nix/store/2ysp5ichpccf4lv1wp2qcwz0bmm840f1-rtmp
nginx>  + ngx_rtmp_module was configured
nginx> adding module in /nix/store/6pb7j6kymf3y4xs5blp3g8mwin2j22kk-dav
nginx>  + ngx_http_dav_ext_module was configured
nginx> adding module in /nix/store/zx05c5lb0ivmjx6fmaa5csjrqpc7fwjc-moreheaders
nginx>  + ngx_http_headers_more_filter_module was configured
nginx> adding module in /nix/store/7d28d91dmzwpf4ahm0wjjy88ajqrwc8c-brotli
nginx> ./configure: error: no /nix/store/7d28d91dmzwpf4ahm0wjjy88ajqrwc8c-brotli/config was found
error: builder for '/nix/store/0aia49f14xb35gqzz9imb7hmviyc2dyr-nginx-1.22.1.drv' failed with exit code 1;

```

Reply:
Weird, it looks like whenever the `fetchFromGitHub` derivation depends on another derivation (here `pkgs.brotli`) the output path at evaluation time doesn't match the built output path:

```
$ nix-instantiate --eval --expr 'with import ./. {}; nginxModules.brotli.src.outPath'
"/nix/store/b81gnvm5f4b5xj4lc2dcif2syb7693if-brotli"
$ nix-build --expr 'with import ./. {}; nginxModules.brotli.src'
/nix/store/dsfqivq4ch12hm1impylix7dnxxlv6gr-brotli
```

Let's leave it be then

Reply:
> Let's leave it be then

Ok.

Reply:
Reported here https://github.com/NixOS/nix/issues/7150#issuecomment-1351889450